MWPW-152581 Extension to promote to use upload report locked status in excel and ui

raga-adbe-gh opened the pull request:
Updates

1. Report as locked when the file is locked / could not be written due to file being open (in Copy or Promote)
2. In parallel waits for all request to be processed before picking next. Using the inparallel (from miloc) which will pick next tasks as and when a previous task completes.
3. Using parts of request wrapper (from miloc) for handling the status code and has option to skip status when required.
4. Timeout  (Status code 600) or Fetch errors (Status code 503) are also retry
5. Locked flag is passed over for reporting
6. Test cases updated

Tests - 
Scenario 1 - Copy from Main Tree to Floodgate Tree retain version
<img width="494" alt="image" src="https://github.com/user-attachments/assets/37ede04b-4bf7-445a-88dd-e1184b6b6120">

Scenario 2 - File is opened and copy is triggered
![image](https://github.com/user-attachments/assets/07b0e3d2-8862-476b-a75a-38ec0d63c9c6)

Scenario 3 - File is locked and copy is triggered
| Reported in GUI | Reported in Excel |
| -------- | -------- |
| <img width="274" alt="image" src="https://github.com/user-attachments/assets/9d4b4a6c-0494-4e8f-a8b9-809ec36b0073"> | ![image](https://github.com/user-attachments/assets/dda6090f-b799-47ff-bcd1-789cfcfe0c4a) | 

Scenario 4 - File is opened (or Locked) and promote is triggered
| Reported in GUI | Reported in Excel |
| -------- | -------- |
| ![image](https://github.com/user-attachments/assets/2741c7f6-d407-43cd-863e-ffeea26210d4) | ![image](https://github.com/user-attachments/assets/4aafab71-2fe7-4716-98ef-85999f0a8d9f) |

Commit:
MWPW-152581 Extension to promote to use upload report locked status in excel and ui

1. Report as locked when the file is locked / could not be written due to lock (in Copy or Promote)
2. In Parallel waits for all request to be processed before picking next. Using the inparallel (from miloc) which will pick next tasks as and when a previous task compltes
3. Using request wrapper (from miloc) for handling the and option to skip status
4. Timeout or Fetch errors are also retry (as internally reported as 503 instead)
5. Locked flag is passed over for reporting

sukamat commented on actions/fgCopyActionHelper.js:
```diff
@@ -54,12 +52,13 @@ class FloodgateActionHelper {
                             // Save the file in the floodgate destination location
                             const saveStatus = await sharepoint.saveFile(file, destination, true);
```
@raga-adbe-gh This is calling the saveFile() which still has the delete file on locked code as I see in `sharepoint.js`. So for copy from source to pink, it is still overwriting but for promote it is creating a new version?

raga-adbe-gh replied:
Thanks @sukamat, Correct for copy flow is unchanged. The content write (new version) is only added for promote when content is moved from pink tree to main tree.

Commit:
Added a space

sukamat commented:
@raga-adbe-gh As discussed this morning, let's update the Copy to use the same upload API that we are using for Promote. The session creation functionality is no longer required I think, so it can be cleaned up or can be left as-is. No strong preference on that.

Commit:
MWPW-152581 Copy from main to floodgate tree to upload

Copy to floodgate tree also upload files instead of copy to retain version.
Opened files will reported as locked

raga-adbe-gh commented:
> @raga-adbe-gh As discussed this morning, let's update the Copy to use the same upload API that we are using for Promote. The session creation functionality is no longer required I think, so it can be cleaned up or can be left as-is. No strong preference on that.

Thanks @sukamat , As suggested updated the flow
